File: thisboot/src/main/java/com/brageast/blog/thisboot/entity/TResult.java
package com.brageast.blog.thisboot.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Data
@AllArgsConstructor
public class TResult<T> {
    private TState result;
    private String message;
    private T data;

    public static <T> Mono<TResult<T>> ofMono(TState result, String message, T data) {
        return Mono.just(new TResult<T>(result, message, data));
    }

    public static <T> Flux<TResult<T>> ofFlux(TState result, String message, T data) {
        return Flux.just(new TResult<T>(result, message, data));
    }
}

File: thisboot/src/main/kotlin/com/brageast/blog/thisboot/entity/TResult.kt
package com.brageast.blog.thisboot.entity


data class TResult<T>(
        val result: TState,
        val message: String,
        var data: T? = null
)

enum class TState constructor(val code: Int) {
    //成功
    SUCCESS(200),

    //失败
    FAIL(400),

    //错误
    FALSE(404),

    //token过期
    EXPIRED(401),

    //无权限
    NOPERMISSION(403),

    //内部错误
    ERROR(500)
}

